pkg/client/unversioned/request.go

```diff
@@ -944,11 +960,12 @@ func retryAfterSeconds(resp *http.Response) (int, bool) {
 
 // Result contains the result of calling Request.Do().
 type Result struct {
-	body       []byte
-	err        error
-	statusCode int
+	body        []byte
+	contentType string
```

Comment:
The returned response will dictate which decoder to use - negotiated
serializer is part of #17922